benchmark/Benchmark/Htoml.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Benchmark.Htoml
+       ( decode
+       , convert
+       , parse
+       ) where
+
+import Benchmark.Type (HaskellType)
+import Control.DeepSeq (NFData, rnf, rwhnf)
+import Data.Aeson.Types (parseEither, parseJSON)
+import Data.Text (Text)
+import GHC.Generics (Generic)
+import Text.Parsec.Error (ParseError)
+
+import Text.Toml (parseTomlDoc)
+import Text.Toml.Types (Node, Table)
+
+import qualified Text.Toml.Types as TTT
+
+
+-- | Decode toml file to Haskell type.
+decode :: Text -> Either String HaskellType
+decode txt = case parseTomlDoc "" txt of
```

Comment:
Well, since you pattern-match on `Right` constructor in the `main` module already, we don't need to have explicit error here. You can just write:

```haskell
decode = parse >=> convert
```

Reply:
Oh, wait. No, you can't, because `Left` constructor has different types for `parse` and `convert` functions. Okay, nevermind.